exporter/exportertest/contract_checker.go

```diff
@@ -0,0 +1,271 @@
+		id := UniqueIDAttrVal(strconv.Itoa(i))
+		fmt.Println("Preparing log number: ", id)
+		data := CreateOneLogWithID(id)
+
+		err = exp.ConsumeLogs(ctx, data)
+	}
+
+	reqCounter := mockReceiver.RequestCounter()
+	// The overall number of requests sent by exporter
+	fmt.Printf("Number of export tries: %d\n", reqCounter.total)
+	// Successfully delivered items
+	fmt.Printf("Total items received successfully: %d\n", reqCounter.success)
+	// Number of errors that happened
+	fmt.Printf("Number of permanent errors: %d\n", reqCounter.error.permanent)
+	fmt.Printf("Number of non-permanent errors: %d\n", reqCounter.error.nonpermanent)
+
+	assert.EventuallyWithT(params.T, func(c *assert.CollectT) {
+		checkIfTestPassed(params.T, params.NumberOfTestElements, reqCounter)
+	}, 2*time.Second, 100*time.Millisecond)
+}
+
+// Test is successful if all the elements were received successfully and no error was returned
+func alwaysSucceedsPassed(t *testing.T, allRecordsNumber int, reqCounter RequestCounter) {
+	require.Equal(t, allRecordsNumber, reqCounter.success)
+	require.Equal(t, reqCounter.total, allRecordsNumber)
+	require.Equal(t, reqCounter.error.nonpermanent, 0)
+	require.Equal(t, reqCounter.error.permanent, 0)
+}
+
+// Test is successful if all the elements were retried on non-permanent errors
+func randomNonPermanentErrorConsumeDecisionPassed(t *testing.T, allRecordsNumber int, reqCounter RequestCounter) {
+	// more or equal tries than successes
+	require.GreaterOrEqual(t, reqCounter.total, reqCounter.success)
+	// it is retried on every error
+	require.Equal(t, reqCounter.total-reqCounter.error.nonpermanent, reqCounter.success)
+	require.Equal(t, allRecordsNumber+reqCounter.error.nonpermanent, reqCounter.total)
+}
+
+// Test is successful if the calls are not retried on permanent errors
+func randomPermanentErrorConsumeDecisionPassed(t *testing.T, allRecordsNumber int, reqCounter RequestCounter) {
+	require.Equal(t, allRecordsNumber-reqCounter.error.permanent, reqCounter.success)
+	require.Equal(t, reqCounter.total, allRecordsNumber)
+}
+
+// Test is successful if the calls are not retried on permanent errors
+func randomErrorConsumeDecisionPassed(t *testing.T, allRecordsNumber int, reqCounter RequestCounter) {
+	require.Equal(t, allRecordsNumber-reqCounter.error.permanent, reqCounter.success)
+	require.Equal(t, reqCounter.total, allRecordsNumber+reqCounter.error.nonpermanent)
+}
+
+func CreateOneLogWithID(id UniqueIDAttrVal) plog.Logs {
+	data := plog.NewLogs()
+	data.ResourceLogs().AppendEmpty().ScopeLogs().AppendEmpty().LogRecords().AppendEmpty().Attributes().PutStr(
+		UniqueIDAttrName,
+		string(id),
+	)
+	return data
+}
+
+func CreateOneTraceWithID(id UniqueIDAttrVal) ptrace.Traces {
+	data := ptrace.NewTraces()
+	data.ResourceSpans().AppendEmpty().ScopeSpans().AppendEmpty().Spans().AppendEmpty().Attributes().PutStr(
+		UniqueIDAttrName,
+		string(id),
+	)
+	return data
+}
+
+func CreateOneMetricWithID(id UniqueIDAttrVal) pmetric.Metrics {
```

Comment:
Should not be exposed. Limiting the API should the primary goal. Otherwise, it'll be hard to maintain and change after 1.0 release

Reply:
changes, I think nothing left is uppercase unnecessarily